nibabel/filebasedimages.py

```diff
@@ -511,3 +512,89 @@ def path_maybe_image(klass, filename, sniff=None, sniff_max=1024):
         if sniff is None or len(sniff[0]) < klass._meta_sniff_len:
             return False, sniff
         return klass.header_class.may_contain_header(sniff[0]), sniff
+
+
+class SerializableImage(FileBasedImage):
+    '''
+    Abstract image class for (de)serializing images to/from byte strings.
+
+    The class doesn't define any image properties.
+
+    It has:
+
+    methods:
+
+       * to_bytes() - serialize image to byte string
+
+    classmethods:
+
+       * from_bytes(bytestring) - make instance by deserializing a byte string
+
+    Loading from byte strings should provide round-trip equivalence:
+
+    .. code:: python
+
+        img_a = klass.from_bytes(bstr)
+        img_b = klass.from_bytes(img_a.to_bytes())
+
+        np.allclose(img_a.get_fdata(), img_b.get_fdata())
+        np.allclose(img_a.affine, img_b.affine)
+
+    Further, for images that are single files on disk, the following methods of loading
+    the image must be equivalent:
+
+    .. code:: python
+
+        img = klass.from_filename(fname)
+
+        with open(fname, 'rb') as fobj:
+            img = klass.from_bytes(fobj.read())
+
+    And the following methods of saving a file must be equivalent:
+
+    .. code:: python
+
+        img.to_filename(fname)
+
+        with open(fname, 'wb') as fobj:
+            fobj.write(img.to_bytes())
+
+    Images that consist of separate header and data files (e.g., Analyze
+    images) currently do not support this interface.
```

Comment:
Maybe say something about the fact that image types can override this class method to implement it, for multi-file images.